spec/requests/api/v1/merchants_request_spec.rb

```diff
@@ -1,30 +1,138 @@
+            expect(merchant_response["id"]).to eq(1)
+            expect(merchant_response["name"]).to eq("MerchantName")
+          end
+        end
+
+        context "based on id" do
+          let(:params) { "id=1" }
+          it_behaves_like "a response that finds a single merchant"
+        end
+
+        context "based on name" do
+          let(:params) { "name=MerchantName" }
+          it_behaves_like "a response that finds a single merchant"
+        end
+
+        context "based on created_at" do
+          let(:params) { "created_at=2012-03-06T16:54:31" }
+          it_behaves_like "a response that finds a single merchant"
+        end
+
+        context "based on updated_at" do
+          let(:params) { "updated_at=2013-03-06T16:54:31" }
+          it_behaves_like "a response that finds a single merchant"
+        end
+      end
+
+      describe "find_all?" do
+        subject { get "/api/v1/merchants/find_all?#{params}" }
+        let(:merchant_response) { JSON.parse(response.body) }
+
+        before do
+          create :merchant, id: 1,
+                            name: "DifferentMerchant",
+                            created_at: "2014-03-06T16:54:31",
+                            updated_at: "2015-03-06T16:54:31"
+          create :merchant, id: 2,
+                            name: "SameMerchant",
+                            created_at: "2013-03-06T16:54:31",
+                            updated_at: "2014-03-06T16:54:31"
+          create :merchant, id: 3,
+                            name: "SameMerchant",
+                            created_at: "2013-03-06T16:54:31",
+                            updated_at: "2014-03-06T16:54:31"
+        end
+
+        shared_examples_for 'a response that finds merchants' do |*merchant_ids|
+          it "finds the correct merchants" do
+            subject
+            expect(response).to be_success
+            expect(merchant_response).to be_an Array
+            expect(merchant_response.map { |result| result['id'] }).to contain_exactly(*merchant_ids)
+          end
+        end
 
-  it "can get single merchant by id" do
-    id = create(:merchant).id
+        context 'based on id' do
+          let(:params) { "id=2" }
+          it_behaves_like 'a response that finds merchants', 2
+        end
 
-    get "/api/v1/merchants/#{id}"
+        context "based on name" do
+          let(:params) { "name=SameMerchant" }
+          it_behaves_like 'a response that finds merchants', 2, 3
+        end
 
-    merchant = JSON.parse(response.body)
+        context "based on created_at" do
+          let(:params) { "created_at=2013-03-06T16:54:31" }
+          it_behaves_like 'a response that finds merchants', 2, 3
+        end
 
-    expect(response).to be_success
-    expect(merchant["id"]).to eq(id)
+        context "based on updated_at" do
+          let(:params) { "updated_at=2014-03-06T16:54:31" }
+          it_behaves_like 'a response that finds merchants', 2, 3
```

Comment:
I haven't seen the behaves_like before, could you give me a brief summary of what's going on?